lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
+      self.maxsize = maxsize
+      self._init(maxsize)
```

Comment:
"I'm" not.  If you look in the Python Queue module that is an exact copy paste of the Queue objects init with self.mutex=threading.Lock changed to self.mutex=threading.RLock.